util/getUniqueCurrentProjects.js

```diff
@@ -0,0 +1,24 @@
+export default function getUniqueCurrentProjects(teamMembers) {
+  const teamMemberIdArray = Object.keys(teamMembers);
+  const uniqueCurrentProjects = [];
+
+  for (let i = 0; i < teamMemberIdArray.length; i += 1) {
+    for (let j = 0; j < teamMembers[teamMemberIdArray[i]].currentProjects.length; j += 1) {
+      const thisProject = teamMembers[teamMemberIdArray[i]].currentProjects[j].project;
+
+      if (!uniqueCurrentProjects.includes(thisProject)) {
+        uniqueCurrentProjects.push(thisProject);
+      }
+    }
+  }
```

Comment:
I think we can simplify this a bit by using [flat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) and [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#remove_duplicate_elements_from_the_array).

```js
const projectsList = Object.keys(teamMembers).map((key) => teamMembers[key].currentProjects);

const uniqueProjects = [...new Set(projectsList.flat())];
```

This should remove duplicates and _may_ sort it automatically, but definitely test it out before removing the sorting methods.

Reply:
TIL: `Set`
TIL: `.flat()`